openmc/tallies.py

```diff
@@ -1482,19 +1482,30 @@ def _outer_product(self, other, binary_op):
             new_name = '({0} {1} {2})'.format(self.name, binary_op, other.name)
             new_tally.name = new_name
 
+        # Create copies of self and other tallies to rearrange for tally
+        # arithmetic
+        self_copy = copy.deepcopy(self)
+        other_copy = copy.deepcopy(other)
```

Comment:
That's what I thought, this should be fine, I like this behavior more than the current implementation. Plus it gives us more freedom to manipulate the copied tallies (if ever needed for future extensions of tally arithmetic) without affecting the user's later use of the arithmetic operands.